### DISCUSSION TITLE
Twitter Authentication

### DISCUSSION THREAD

**Discussion by alexluong** [sergiodxa/remix-auth#114] [Ideas]
Hi, I started working with Remix recently and would love to use `remix-auth`. I'm using Twitter OAuth 1.0 for authentication, just wondering if there's any plan to support it in the future.

I'd be open to maybe contributing as well, would love to get feedback first and any pointers on where to start would be nice.

Thanks!

**Comment by sergiodxa** [comment]
@alexluong strategies are published in individual packages, see #111 for some strategies already built by the community, I remember a few people were trying to build a strategy for Twitter but no one has published one yet, you can try to build it and publish it.

**Reply by alexluong** [reply]
Gotcha, sounds good. I'll check that out!

**Comment by alexluong** [comment]
So I spent the last day of 2021 looking into this. Woo! OAuth 1.0, amirite...

Anyway, I'm definitely interested in doing this, but afraid I don't have the time right now. Just sharing my research in case anybody is interested in taking it over.

Twitter auth uses OAuth 1.0, which is different than the other popular social auth strategies. It's kind of a pain tbh.

Some relevant links I found:

- [official docs from Twitter](https://developer.twitter.com/en/docs/authentication/oauth-1-0a/obtaining-user-access-tokens)
- [`passport-oauth1`](https://github.com/jaredhanson/passport-oauth1) which is probably something I'll try to work off of if I were to do this
- [`twittersignin`](https://github.com/shalvah/twittersignin) which is another implementation on top of [Twit](https://www.npmjs.com/package/twit), so a little outdated but probably still works fine, can be a good resource to consult as well

**Comment by MichaelDeBoey** [comment]
@alexluong I think you should take a look at @TheRealFlyingCoder's [`remix-auth-socials`](https://github.com/TheRealFlyingCoder/remix-auth-socials) package, as that has Twitter support planned

**Reply by alexluong** [reply]
Thanks for sharing. It's definitely on my radar, curious to see when it's gonna be supported!

**Comment by na2hiro** [comment]
I published [`remix-auth-twitter`](https://github.com/na2hiro/remix-auth-twitter) right now. It was hard due to outdated Twitter documentation.

Feedback welcome! And Thanks @sergiodxa for the great abstraction and the template. I could focus on the core task.

**Reply by alexluong** [reply]
Thanks, it looks really good, will try to incorporate it when I've got around to that project and will report back how that goes.
